pycbc/frame/gwosc.py

```diff
@@ -0,0 +1,185 @@
+        The GPS time.
+    ifo: str
+        The interferometer prefix string. Optional and normally unused,
+        except for some special times where data releases were made for a
+        single detector under unusual circumstances. For example, to get
+        the data around GW170608 in the Hanford detector.
+    """
+    cases = [
+        (
+            # ifo is only needed in this special case, otherwise,
+            # the run name is the same for all ifos
+            1180911618 <= time <= 1180982427 and ifo == 'H1',
+            'BKGW170608_16KHZ_R1'
+        ),
+        (1253977219 <= time <= 1320363336, 'O3b_16KHZ_R1'),
+        (1238166018 <= time <= 1253977218, 'O3a_16KHZ_R1'),
+        (1164556817 <= time <= 1187733618, 'O2_16KHZ_R1'),
+        (1126051217 <= time <= 1137254417, 'O1'),
+        (815011213 <= time <= 875318414, 'S5'),
+        (930787215 <= time <= 971568015, 'S6')
+    ]
+    for condition, name in cases:
+        if condition:
+            return name
+    raise ValueError(f'Time {time} not available in a public dataset')
+
+
+def _get_channel(time):
+    if time < 1164556817:
+        return 'LOSC-STRAIN'
+    return 'GWOSC-16KHZ_R1_STRAIN'
+
+
+def gwosc_frame_json(ifo, start_time, end_time):
+    """Get the information about the public data files in a duration of time.
+
+    Parameters
+    ----------
+    ifo: str
+        The name of the interferometer to find the information about.
+    start_time: int
+        The start time in GPS seconds.
+    end_time: int
+        The end time in GPS seconds.
+
+    Returns
+    -------
+    info: dict
+        A dictionary containing information about the files that span the
+        requested times.
+    """
+    run = get_run(start_time)
+    run2 = get_run(end_time)
+    if run != run2:
+        raise ValueError(
+            'Spanning multiple runs is not currently supported. '
+            f'You have requested data that uses both {run} and {run2}'
+        )
+
+    url = _GWOSC_URL % (run, ifo, int(start_time), int(end_time))
+
+    try:
+        return json.loads(urlopen(url).read().decode())
+    except Exception as exc:
+        msg = ('Failed to find gwf files for '
+               f'ifo={ifo}, run={run}, between {start_time}-{end_time}')
+        raise ValueError(msg) from exc
+
+
+def gwosc_frame_urls(ifo, start_time, end_time):
+    """Get a list of URLs to GWOSC frame files.
+
+    Parameters
+    ----------
+    ifo: str
+        The name of the interferometer to find the information about.
+    start_time: int
+        The start time in GPS seconds.
+    end_time: int
+        The end time in GPS seconds.
+
+    Returns
+    -------
+    frame_files: list
+        A dictionary containing information about the files that span the
+        requested times.
+    """
+    data = gwosc_frame_json(ifo, start_time, end_time)['strain']
+    return [d['url'] for d in data if d['format'] == 'gwf']
+
+
+def read_frame_gwosc(channels, start_time, end_time):
```

Comment:
Would it be worth setting an alias to losc  with a deprecation warning? I'll let  you decide on that.

Reply:
I might open a new PR for that.